sokutei.h

```diff
@@ -0,0 +1,194 @@
+                                ? sizeof(SOKUTEI_INTEGER_COUNTER_TYPE) \
+                                : sizeof(SOKUTEI_INTERVAL_TIMER_COUNTER_TYPE))\
+                            : ((sizeof(SOKUTEI_FLOAT_COUNTER_TYPE) > sizeof(SOKUTEI_INTERVAL_TIMER_COUNTER_TYPE)) \
+                                ? sizeof(SOKUTEI_FLOAT_COUNTER_TYPE) \
+                                : sizeof(SOKUTEI_INTERVAL_TIMER_COUNTER_TYPE)))
+
+#define SOKUTEI_INTEGER_TYPE '1'
+#define SOKUTEI_FLOAT_TYPE '2'
+#define SOKUTEI_INTERVAL_TYPE '3'
+
+#define SOKUTEI_NOT_FOUND -1
+#define SOKUTEI_NOT_MATCHING_TYPE -2
+#define SOKUTEI_COUNTER_LIMIT_REACHED -3
+
+
+#ifdef SOKUTEI_LOGGING_ENABLED
+#define SOKUTEI_DEBUG_PRINT(x, y) printf(x, y)
+#else
+#define SOKUTEI_DEBUG_PRINT(x, y)
+#endif
+
+
+int sokutei_strcmp(const char *string_a, const char *string_b){
+    int i = 0;
+    while(string_a[i] == string_b[i] && string_a[i]) i++;
+    return string_a[i] - string_b[i];
+}
+
+
+char *sokutei_strcpy(char *string_a, const char *string_b){
+    int i;
+    for(i = 0; (string_a[i] = string_b[i]) && (i < SOKUTEI_MAX_COUNTER_NAME_LENGTH); i++);
+    string_a[SOKUTEI_MAX_COUNTER_NAME_LENGTH] = '\0';
+    return string_a;
+}
+#define sokutei_string_equals(string_a, string_b) (sokutei_strcmp(string_a, string_b) == 0)
+
+
+int sokutei_number_of_iterations = SOKUTEI_MAX_MEASURED_ITERATIONS;
+
+/**
+ * sokutei_counter_definitions is an array consist of the counter name followed by string separator (\0), and the type of the counter
+ */
+char sokutei_counter_definitions[SOKUTEI_MAX_COUNTER_COUNT][SOKUTEI_MAX_COUNTER_NAME_LENGTH + 2] = {'\0'};
+
+
+char sokutei_counters[SOKUTEI_MAX_COUNTER_COUNT * MAX_SIZE_OF_TYPES * SOKUTEI_MAX_MEASURED_ITERATIONS] = {0};
+int  sokutei_number_of_counters = 0;
+int number_of_iterations = SOKUTEI_MAX_MEASURED_ITERATIONS;
+
+
+char sokutei_get_type_of(const int index){
+    return sokutei_counter_definitions[index][31];
+}
+
+
+int sokutei_get_index_of_counter(const char *counter_name){
+    int i;
+    for(i = 0; i < sokutei_number_of_counters; i++){
+        if(sokutei_string_equals(sokutei_counter_definitions[i], counter_name)){
+            return i;
+        }
+    }
+    return SOKUTEI_NOT_FOUND;
+}
+
+
+void sokutei_set_number_of_iterations(const int iterations){
+
+    if(SOKUTEI_MAX_MEASURED_ITERATIONS < iterations){
+        //TODO call error
+        return;
+    }
+
+    number_of_iterations = iterations;
+}
+
+
+int sokutei_add_counter(const char *counter_name, char type){
+
+    if(type != SOKUTEI_INTEGER_TYPE && type != SOKUTEI_FLOAT_TYPE && type != SOKUTEI_INTERVAL_TYPE){
+        return SOKUTEI_NOT_MATCHING_TYPE;
+    }
+
+    if(sokutei_number_of_counters + 1 >= SOKUTEI_MAX_COUNTER_COUNT){
+        return SOKUTEI_COUNTER_LIMIT_REACHED;
+    }
+
+    sokutei_strcpy(sokutei_counter_definitions[sokutei_number_of_counters], counter_name);
+    sokutei_counter_definitions[sokutei_number_of_counters][SOKUTEI_MAX_COUNTER_NAME_LENGTH + 1] = type;
```

Comment:
If you were to use a constant for that `31` magic number, you'd be able to make use of it here as well.